### PR TITLE
#1476 Add notes to program enrolments

### LINKED ISSUE #1476: Add a Note section to all program enrolment forms
Should be a generic field for ALL enrolment forms (for any country) to have the possibility to write notes at the enrolment level.

Ideally, they should be the same as for the patient registration form: 

Screen Shot 2023-03-31 at 5 15 33 PM
 
Otherwise, a simple "Notes" free text field should be fine.

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fixes #1476

# 👩🏻‍💻 What does this PR do? 
 

Just updates the schemas

# 🧪 How has/should this change been tested? 

Enrol a patient in both "HIV Care and Treatment" and "HIV Testing" programs -- check there is a Notes section in the form.

### CODE REVIEW

**Comment by roxy-dao** [comment]
Looks great~

🤔 You might want to update this with 1463 once it's been merged since there are some BE changes there that are required here to run.

Also, does the `Routine Immunisation` form need an array of notes as well, or is a single note text field alright for this form? 🤔

File: server/service/src/sync/program_schemas/hiv_care_program.json
Comment (mark-prins) [inline server/service/src/sync/program_schemas/hiv_care_program.json:203]: no mention in the PR or issue of why the required fields might need to be dropped. Is there a reason?

File: server/service/src/sync/program_schemas/hiv_care_program.json
Comment (mark-prins) [inline server/service/src/sync/program_schemas/hiv_care_program.json:305]: where has this come in from?

File: server/service/src/sync/program_schemas/hiv_testing_program.json
Comment (mark-prins) [inline server/service/src/sync/program_schemas/hiv_testing_program.json:57]: have these come from somewhere? some descriptions sound like developer comments 😉

**Review by mark-prins — CHANGES_REQUESTED** [review]
I'm unable to compile the server, without making the changes noted here:

```
Compiling service v0.1.0 (/Users/m2/GitHub/open-msupply-pr/server/service)
error[E0063]: missing field `notes` in initializer of `HivtestingProgramEnrolment`
  --> service/src/sync/init_programs_data.rs:41:13
   |
41 |             Self {
   |             ^^^^ missing `notes`

error[E0063]: missing fields `notes` and `previous_clinic_art_id` in initializer of `HivcareProgramEnrolment`
  --> service/src/sync/init_programs_data.rs:60:13
   |
60 |             Self {
   |             ^^^^ missing `notes` and `previous_clinic_art_id`

For more information about this error, try `rustc --explain E0063`.
error: could not compile `service` due to 2 previous errors
```

**Comment by mark-prins** [comment]
heh. I don't see that we can avoid it - adding the notes has highlighted some modal-on-modal action 🤷 

image

File: server/service/src/sync/program_schemas/hiv_care_program.json
Comment (CarlosNZ) [reply server/service/src/sync/program_schemas/hiv_care_program.json:203]: All these schema changes are just imported from the latest version in the program schema repo. (I just re-exported them).

This one in particular (and maybe the others) are from what @clemens-msupply  did in #1463, which hasn't been merged yet. It's possible the other changes are from PRs that also aren't merged yet.

I'll co-ordinate with @clemens-msupply in the morning to clarify and make sure we merge in the correct order (so as to not end up with compile errors like above)

**Comment by mark-prins** [comment]
sorry - one more question. 
The 'routine immunisation' form is different. that's ok?

It has a single-line-notes field, and also has a tab which isn't translating for me. I've recently changed the `ModalTabs` component in develop: to pass in a LocaleKey rather than a string which then gets retranslated. Can't see the changes in this branch yet.. but when they come in, could we update the ProgramEnrolment/DetailModal?

image

File: server/service/src/sync/program_schemas/hiv_testing_program.json
Comment (CarlosNZ) [reply server/service/src/sync/program_schemas/hiv_testing_program.json:57]: They're auto-generated by the programschemas repo based on comments. Won't show up in the UI.

File: server/service/src/sync/program_schemas/hiv_care_program.json
Comment (CarlosNZ) [reply server/service/src/sync/program_schemas/hiv_care_program.json]: Okay, the other schema diffs are gone now after a fresh merge from feature branch. Only this one remains and it's from #1463. So I won't merge this one until after that.

**Comment by CarlosNZ** [comment]
> The 'routine immunisation' form is different. that's ok?

I've updated it to use the same Notes: a8eba45ad (as well as fixed another little discrepancy in that program)

**Comment by CarlosNZ** [comment]
> and also has a tab which isn't translating for me.

Screenshot 2023-04-05 at 10 17 28 AM

Seems okay now (it's the same form as the other two programs)

File: server/service/src/sync/program_schemas/hiv_testing_program.json
Comment (clemens-msupply) [reply server/service/src/sync/program_schemas/hiv_testing_program.json:57]: We experimented with showing them as help text in the UI though... But in this case there is a special component that renders the note anyway.

**Review by roxy-dao — APPROVED** [review]
Thanks Carl~

**Review by clemens-msupply — APPROVED** [review]
Works great thank you!

**Pull request merged**
